### PR TITLE
[test] Add test cases for 'BATCH_TO_SPACE_ND' and 'TRANSPOSE' op

### PR DESCRIPTION
**Pull request by cuiyanx**
Add 37 test cases for "BATCH_TO_SPACE_ND" and "TRANSPOSE" op:

op | cts | plus | test-Model.js
-- | -- | -- | --
BATCH_TO_SPACE_ND | 2 | 0 | 16
TRANSPOSE | 2 | 0 | 17

Fix: #579 and #581 
Signed-off-by: cuiyan <yanx.cui@intel.com>

### CODE REVIEW

**Comment by cuiyanx** [comment]
@BruceDai please review

**Comment by cuiyanx** [comment]
Add 38 test cases for "BATCH_TO_SPACE_ND" and "TRANSPOSE" op:

op | cts | plus | test-Model.js
-- | -- | -- | --
BATCH_TO_SPACE_ND | 2 | 0 | 17
TRANSPOSE | 2 | 0 | 17

**Comment by cuiyanx** [comment]
@BruceDai please review

File: test/unit-test/test-Model.js [hunk 1]
```diff
@@ -6836,6 +6836,596 @@ describe('Unit Test/Model Test', function() {
         });
       });
     });
+
+    it('"4-D tensor as input0 that is TENSOR_FLOAT32 type and 1-D tensor as input1 that is TENSOR_INT32 type" are ok for "BATCH_TO_SPACE_ND" operation', function() {
```
Comment (BruceDai) [inline test/unit-test/test-Model.js]: @cuiyanx According to https://github.com/intel/webml-polyfill/blob/master/src/nn/Enums.js#L1199, **All values of 1-D Tensor must be >= 1.** , we should add this info into test case purpose.

File: test/unit-test/test-Model.js [hunk 2]
```diff
@@ -6836,6 +6836,596 @@ describe('Unit Test/Model Test', function() {
...
+        model.addOperand(input0);
+        model.addOperand(input1);
+        model.addOperand(output);
+        assert.throws(() => {
+          model.addOperation(nn.BATCH_TO_SPACE_ND, [0, 1], [2]);
+        });
+      });
+    });
+
+    it('raise error when 2-D tensor as input0 that is TENSOR_FLOAT32 type and 1-D tensor as input1 that is TENSOR_INT32 type for "BATCH_TO_SPACE_ND" operation', async function() {
```
Comment (BruceDai) [inline test/unit-test/test-Model.js]: Please add one test case with input0 as a Rank 5 tensor.

File: test/unit-test/test-Model.js [hunk 3]
```diff
@@ -6836,6 +6836,596 @@ describe('Unit Test/Model Test', function() {
...
+      await compilation.finish();
+
+      let execution = await compilation.createExecution();
+      execution.setInput(0, new Float32Array([1, 2, 3, 4]));
+      await assertThrowsAsync(async() => {
+        await execution.startCompute();
+      });
+    });
+
+    it('raise error when the block value < 1 for "BATCH_TO_SPACE_ND" operation', async function() {
```
Comment (BruceDai) [inline test/unit-test/test-Model.js]: This test case is for input1 isn't of TENSOR_INT32 type, please add one test case with  input1 as TENSOR_INT32 tensor and its' value < 1.

File: test/unit-test/test-Model.js [hunk 4]
```diff
@@ -6836,6 +6836,596 @@ describe('Unit Test/Model Test', function() {
...
+      await compilation.finish();
+
+      let execution = await compilation.createExecution();
+      execution.setInput(0, new Float32Array([1, 2, 3, 4]));
+      execution.setOutput(0, new Float32Array(product(output.dimensions)));
+      await assertThrowsAsync(async() => {
+        await execution.startCompute();
+      });
+    });
+
```
Comment (BruceDai) [inline test/unit-test/test-Model.js]: Please add one test case with 3 inputs (1 output) ,  and  one test case with 2 outputs (2 inputs), the similar test cases for TRANSPOSE op

File: test/unit-test/test-Model.js [hunk 5]
```diff
@@ -6836,6 +6836,596 @@ describe('Unit Test/Model Test', function() {
...
+    it('"4-D tensor as input0 that is TENSOR_FLOAT32 type and 1-D tensor as input1 that is TENSOR_INT32 type" are ok for "TRANSPOSE" operation', function() {
```
Comment (BruceDai) [inline test/unit-test/test-Model.js]: According to https://github.com/intel/webml-polyfill/blob/master/src/nn/Enums.js#L1224,
```
1: An optional 1-D Tensor of {@link TENSOR_INT32}, the permutation of  the dimensions of the input tensor. 
```
input1 is an optional tensor, 
so please add some cases with input having only input0 for TRANSPOSE op, thanks.

**Comment by cuiyanx** [comment]
Add 52 test cases for "BATCH_TO_SPACE_ND" and "TRANSPOSE" op:


op | cts | plus | test-Model.js
-- | -- | -- | --
BATCH_TO_SPACE_ND | 2 | 0 | 19
TRANSPOSE | 2 | 0 | 29


@BruceDai please review

**Comment by BruceDai** [comment]
LGTM, thanks @cuiyanx 
@ibelem please help merge this PR, thanks.

**Pull request merged**